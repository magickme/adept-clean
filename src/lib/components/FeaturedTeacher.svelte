<script lang="ts">
    import type { FeaturedTeacherProps } from '../types/FeaturedTeacher';
    import { getFeaturedImageProps, getOptimizedImageUrl } from '$lib/utils/imageUtils';

    const props = $props();
    const { 
      title = 'Meet Our Team Member', 
      description,
      dataTestid,
      memberName,
      memberRole,
      memberBio,
      memberImage
    } = props;

    const imageProps = getFeaturedImageProps(false);
</script>

{#snippet memberCard()}
    <li class="bg-kohl/30 rounded-lg overflow-hidden">
      <div class="aspect-[3/2] w-full">
        <img 
          class="h-full w-full object-cover object-[center_33%]" 
          src={getOptimizedImageUrl(memberImage, 800)} 
          alt={memberName}
          {...imageProps}
        >
      </div>
      <div class="p-4 sm:p-6">
        <h3 class="text-xl sm:text-2xl font-semibold leading-8 text-ivory font-montserrat">{memberName}</h3>
        <p class="mt-2 text-base leading-7 text-ivory/70 font-sourceSans">{memberRole}</p>
        <div class="mt-4 text-sm sm:text-base leading-7 text-ivory/70 font-sourceSans space-y-4">
          {@html memberBio}
        </div>
        <a
          href="https://www.jasonlouv.com"
          target="_blank"
          rel="noopener noreferrer"
          class="mt-6 w-full sm:w-auto px-4 py-2 bg-cerulean text-ivory rounded-md hover:bg-corsair transition-colors duration-300 font-sourceSans inline-block text-center"
          aria-label="Visit Jason Louv's personal website to learn more about his work and teachings"
        >
          Visit Jason Louv's Website
        </a>
      </div>
    </li>
{/snippet}
  
<div class="bg-kohl py-12 sm:py-16 md:py-24" data-testid={dataTestid}>
  <div class="mx-auto grid max-w-7xl grid-cols-1 gap-x-6 gap-y-8 px-4 sm:px-6 lg:px-8 xl:grid-cols-3">
    <div class="mx-auto max-w-2xl lg:mx-0">
      <h2 class="text-2xl sm:text-3xl font-bold tracking-tight text-cerulean font-montserrat">{@html title}</h2>
      <div class="mt-4 text-base sm:text-lg leading-8 text-ivory/70 font-sourceSans">
        {@html description}
      </div>
    </div>
    <ul role="list" class="mx-auto grid max-w-2xl grid-cols-1 gap-4 sm:gap-6 sm:grid-cols-2 lg:mx-0 lg:max-w-none lg:gap-8 xl:col-span-2">
      {@render memberCard()}
    </ul>
  </div>
</div>