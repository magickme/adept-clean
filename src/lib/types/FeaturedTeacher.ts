export interface FeaturedTeacherProps {
  title?: string;
  description?: string;
  dataTestid?: string;
  memberName: string;
  memberRole: string;
  memberBio: string;
  memberImage?: string;
  memberXLink?: string;
  memberLinkedinLink?: string;
}

export interface SelectedMember {
  name: string;
  role: string;
  bio: string;
}

export interface ModalProps {
  title: string;
  message: string;
  buttonText: string;
  open: boolean;
  onClose: () => void;
} 