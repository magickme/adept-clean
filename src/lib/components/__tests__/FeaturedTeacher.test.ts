import { expect, test, describe, beforeEach, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/svelte/svelte5';
import FeaturedTeacher from '../FeaturedTeacher.svelte';
import type { FeaturedTeacherProps } from '../../types/FeaturedTeacherher';

describe('FeaturedTeacher', () => {
  const defaultProps: FeaturedTeacherProps = {
    title: 'Meet Our Team Member',
    description: 'Team description',
    dataTestid: 'featured-teacher',
    memberName: 'John Doe',
    memberRole: 'Senior Teacher',
    memberBio: 'Teacher bio',
    memberImage: '/path/to/image.jpg',
    memberXLink: 'https://x.com/johndoe',
    memberLinkedinLink: 'https://linkedin.com/in/johndoe'
  };

  // Basic Rendering
  test('renders with default props', () => {
    const { container } = render(FeaturedTeacher, { props: defaultProps });
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('featured-teacher')).toBeInTheDocument();
  });

  // Props Validation
  test('renders with minimal props', () => {
    const minimalProps = {
      memberName: 'John Doe',
      memberRole: 'Teacher',
      memberBio: 'Bio'
    };
    render(FeaturedTeacher, { props: minimalProps });
    expect(screen.getByText('John Doe')).toBeInTheDocument();
  });

  // Social Links
  test('renders social links when provided', () => {
    render(FeaturedTeacher, { props: defaultProps });
    expect(screen.getByRole('link', { name: /x/i })).toHaveAttribute('href', defaultProps.memberXLink);
    expect(screen.getByRole('link', { name: /linkedin/i })).toHaveAttribute('href', defaultProps.memberLinkedinLink);
  });

  test('does not render social links when not provided', () => {
    const propsWithoutSocial = { ...defaultProps, memberXLink: undefined, memberLinkedinLink: undefined };
    render(FeaturedTeacher, { props: propsWithoutSocial });
    expect(screen.queryByRole('link', { name: /x/i })).not.toBeInTheDocument();
    expect(screen.queryByRole('link', { name: /linkedin/i })).not.toBeInTheDocument();
  });

  // Modal Functionality
  describe('Modal', () => {
    beforeEach(() => {
      render(FeaturedTeacher, { props: defaultProps });
    });

    test('opens modal on view details click', async () => {
      const viewDetailsBtn = screen.getByTestId('view-details-button');
      await fireEvent.click(viewDetailsBtn);
      
      const dialog = screen.getByRole('dialog');
      const modalTitle = screen.getByTestId('modal-title');
      
      expect(dialog).toBeInTheDocument();
      expect(modalTitle).toHaveTextContent(defaultProps.memberName);
      expect(screen.getByText(defaultProps.memberRole)).toBeInTheDocument();
    });

    test('closes modal on close button click', async () => {
      // Open modal
      await fireEvent.click(screen.getByTestId('view-details-button'));
      // Close modal
      await fireEvent.click(screen.getByTestId('modal-close-button'));
      
      expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
    });

    test('closes modal on backdrop click', async () => {
      // Open modal
      await fireEvent.click(screen.getByTestId('view-details-button'));
      // Click backdrop
      const backdrop = screen.getByTestId('modal-backdrop');
      await fireEvent.click(backdrop);
      
      expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
    });
  });

  // Accessibility
  test('meets accessibility requirements', async () => {
    render(FeaturedTeacher, { props: defaultProps });
    
    // Open modal first
    await fireEvent.click(screen.getByTestId('view-details-button'));
    
    const dialog = screen.getByRole('dialog');
    expect(dialog).toHaveAttribute('aria-modal', 'true');
    expect(dialog).toHaveAttribute('aria-labelledby', 'modal-title');
  });

  // Image Handling
  test('renders member image with correct attributes', () => {
    render(FeaturedTeacher, { props: defaultProps });
    const img = screen.getByAltText(defaultProps.memberName);
    
    expect(img).toHaveAttribute('src', defaultProps.memberImage);
    expect(img).toHaveClass('aspect-[3/2]', 'w-full', 'rounded-2xl', 'object-cover');
  });
}); 