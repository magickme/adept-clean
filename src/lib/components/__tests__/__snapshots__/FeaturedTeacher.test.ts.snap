// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeaturedTeacher > renders with default props 1`] = `
<div>
  <div
    class="bg-kohl py-24 md:py-32 lg:py-40"
    data-testid="featured-teacher"
  >
    <div
      class="mx-auto grid max-w-7xl grid-cols-1 gap-x-8 gap-y-20 px-6 lg:px-8 xl:grid-cols-3"
    >
      <div
        class="mx-auto max-w-2xl lg:mx-0"
      >
        <h2
          class="text-3xl font-bold tracking-tight text-cerulean sm:text-4xl font-montserrat"
        >
          Meet Our Team Member
        </h2>
         
        <p
          class="mt-6 text-lg leading-8 text-ivory/70 font-sourceSans"
        >
          Team description
        </p>
      </div>
       
      <ul
        class="mx-auto grid max-w-2xl grid-cols-1 gap-x-6 gap-y-20 sm:grid-cols-2 lg:mx-0 lg:max-w-none lg:gap-x-8 xl:col-span-2"
        role="list"
      >
        <li>
          <img
            alt="John Doe"
            class="aspect-[3/2] w-full rounded-2xl object-cover"
            src="/path/to/image.jpg"
          />
           
          <h3
            class="mt-6 text-lg font-semibold leading-8 text-ivory font-montserrat"
          >
            John Doe
          </h3>
           
          <p
            class="text-base leading-7 text-ivory/70 font-sourceSans"
          >
            Senior Teacher
          </p>
           
          <p
            class="mt-4 text-base leading-7 text-ivory/70 font-sourceSans space-y-4"
          >
            Teacher bio
            <!---->
          </p>
           
          <ul
            class="mt-6 flex gap-x-6"
            role="list"
          >
            <li>
              <a
                class="text-flannel hover:text-cerulean"
                href="https://x.com/johndoe"
              >
                <span
                  class="sr-only"
                >
                  X
                </span>
                 
                <svg
                  aria-hidden="true"
                  class="h-5 w-5"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                >
                  <path
                    d="M11.4678 8.77491L17.2961 2H15.915L10.8543 7.88256L6.81232 2H2.15039L8.26263 10.8955L2.15039 18H3.53159L8.87581 11.7878L13.1444 18H17.8063L11.4675 8.77491H11.4678ZM9.57608 10.9738L8.95678 10.0881L4.02925 3.03974H6.15068L10.1273 8.72795L10.7466 9.61374L15.9156 17.0075H13.7942L9.57608 10.9742V10.9738Z"
                  />
                </svg>
              </a>
            </li>
            <!---->
             
            <li>
              <a
                class="text-flannel hover:text-cerulean"
                href="https://linkedin.com/in/johndoe"
              >
                <span
                  class="sr-only"
                >
                  LinkedIn
                </span>
                 
                <svg
                  aria-hidden="true"
                  class="h-5 w-5"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                >
                  <path
                    clip-rule="evenodd"
                    d="M16.338 16.338H13.67V12.16c0-.995-.017-2.277-1.387-2.277-1.39 0-1.601 1.086-1.601 2.207v4.248H8.014v-8.59h2.559v1.174h.037c.356-.675 1.227-1.387 2.526-1.387 2.703 0 3.203 1.778 3.203 4.092v4.711zM5.005 6.575a1.548 1.548 0 11-.003-3.096 1.548 1.548 0 01.003 3.096zm-1.337 9.763H6.34v-8.59H3.667v8.59zM17.668 1H2.328C1.595 1 1 1.581 1 2.298v15.403C1 18.418 1.595 19 2.328 19h15.34c.734 0 1.332-.582 1.332-1.299V2.298C19 1.581 18.402 1 17.668 1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </a>
            </li>
            <!---->
          </ul>
           
          <button
            class="mt-4 px-4 py-2 bg-cerulean text-ivory rounded-md hover:bg-cerulean-600 transition-colors duration-300 font-sourceSans"
            data-testid="view-details-button"
          >
            View Details
          </button>
        </li>
        <!---->
      </ul>
    </div>
  </div>
   
  <!---->
  
</div>
`;
